CMakeLists.txt

```diff
@@ -146,3 +167,66 @@ target_link_libraries(test_kmeans PRIVATE lib_kmeans fortuno_wrapper common_flag
 add_test(NAME test_assign_points_to_centroids COMMAND test_kmeans "Assign points to centroids")
 add_test(NAME test_update_centroids_no_movement COMMAND test_kmeans "Update centroids - no change")
 add_test(NAME test_points_are_converged COMMAND test_kmeans "Difference in two sets of points")
+
+#[==============================================================================================[
+#                                   Installation                                                #
+#]==============================================================================================]
+
+# Install the target library and create export-set
+install(
+   TARGETS lib_kmeans
+   EXPORT KmeansTargets
+   LIBRARY
+     DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+   ARCHIVE
+     DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+ )
+
+ # Generate and install the export file
+install(EXPORT KmeansTargets
+   FILE KmeansTargets.cmake
+   NAMESPACE kmeans::
+   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/kmeans"
+   )
+
+# Install the module files
+# Note, this must be consistent with cmake/kmeans.pc.in
+install(
+   DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}
+   DESTINATION "${CMAKE_INSTALL_PREFIX}"    #Copies directory ${CMAKE_Fortran_MODULE_DIRECTORY}, to this location
```

Comment:
Install to `/usr/modules`? I guess you-do-you

Reply:
`${CMAKE_INSTALL_PREFIX}` I thought that's the default, which gets overwritten if I configure with

```shell
cmake --fresh -B serial-cmake-build-debug -DCMAKE_BUILD_TYPE=Debug -DMPI=Off -DCMAKE_INSTALL_PREFIX=/Users/alexanderbuccheri/Codes/kmeans/install
```

Reply:
Otherwise, what variable am I supposed to use?

Reply:
It's not the variable it's the path that you should put it in  `/usr/...` have a defined meaning. There is no standard for `modules` so 🤷‍♂️. But use relative path installation.

Of course it should be an editable variable but I haven't got to the options part of the grill